subsys/bluetooth/mesh/light_xyl_srv.c

```diff
@@ -0,0 +1,586 @@
+		return;
+	}
+
+	struct bt_mesh_light_xyl_srv *srv = model->user_data;
+	struct bt_mesh_light_xyl_range_msg set;
+	enum bt_mesh_model_status status_code;
+
+	set.x.min = net_buf_simple_pull_le16(buf);
+	set.x.max = net_buf_simple_pull_le16(buf);
+	set.y.min = net_buf_simple_pull_le16(buf);
+	set.y.max = net_buf_simple_pull_le16(buf);
+
+	if ((set.x.min > set.x.max) || (set.y.min > set.y.max)) {
+		status_code = BT_MESH_MODEL_STATUS_INVALID;
+		goto respond;
+	}
+
+	status_code = BT_MESH_MODEL_SUCCESS;
+	memcpy(&srv->range, &set, sizeof(set));
+	store_state(srv);
+
+	if (srv->handlers->range_update) {
+		srv->handlers->range_update(srv, ctx, &srv->range);
+	}
+
+	(void)bt_mesh_light_xyl_range_pub(srv, NULL, status_code);
+
+respond:
+	if (ack) {
+		range_rsp(model, ctx, status_code);
+	}
+}
+
+static void range_get_handle(struct bt_mesh_model *model,
+			     struct bt_mesh_msg_ctx *ctx,
+			     struct net_buf_simple *buf)
+{
+	if (buf->len != BT_MESH_LIGHT_XYL_MSG_LEN_GET) {
+		return;
+	}
+
+	range_rsp(model, ctx, BT_MESH_MODEL_SUCCESS);
+}
+
+static void range_set_handle(struct bt_mesh_model *model,
+			     struct bt_mesh_msg_ctx *ctx,
+			     struct net_buf_simple *buf)
+{
+	range_set(model, ctx, buf, true);
+}
+
+static void range_set_unack_handle(struct bt_mesh_model *model,
+				   struct bt_mesh_msg_ctx *ctx,
+				   struct net_buf_simple *buf)
+{
+	range_set(model, ctx, buf, false);
+}
+
+const struct bt_mesh_model_op _bt_mesh_light_xyl_srv_op[] = {
+	{ BT_MESH_LIGHT_XYL_GET, BT_MESH_LIGHT_XYL_MSG_LEN_GET,
+	  xyl_get_handle },
+	{ BT_MESH_LIGHT_XYL_SET, BT_MESH_LIGHT_XYL_MSG_MINLEN_SET,
+	  xyl_set_handle },
+	{ BT_MESH_LIGHT_XYL_SET_UNACK, BT_MESH_LIGHT_XYL_MSG_MINLEN_SET,
+	  xyl_set_unack_handle },
+	{ BT_MESH_LIGHT_XYL_TARGET_GET, BT_MESH_LIGHT_XYL_MSG_LEN_GET,
+	  target_get_handle },
+	{ BT_MESH_LIGHT_XYL_DEFAULT_GET, BT_MESH_LIGHT_XYL_MSG_LEN_GET,
+	  default_get_handle },
+	{ BT_MESH_LIGHT_XYL_RANGE_GET, BT_MESH_LIGHT_XYL_MSG_LEN_GET,
+	  range_get_handle },
+	BT_MESH_MODEL_OP_END,
+};
+
+const struct bt_mesh_model_op _bt_mesh_light_xyl_setup_srv_op[] = {
+	{ BT_MESH_LIGHT_XYL_DEFAULT_SET, BT_MESH_LIGHT_XYL_MSG_LEN_DEFAULT,
+	  default_set_handle },
+	{ BT_MESH_LIGHT_XYL_DEFAULT_SET_UNACK,
+	  BT_MESH_LIGHT_XYL_MSG_LEN_DEFAULT, default_set_unack_handle },
+	{ BT_MESH_LIGHT_XYL_RANGE_SET, BT_MESH_LIGHT_XYL_MSG_LEN_RANGE_SET,
+	  range_set_handle },
+	{ BT_MESH_LIGHT_XYL_RANGE_SET_UNACK,
+	  BT_MESH_LIGHT_XYL_MSG_LEN_RANGE_SET, range_set_unack_handle },
+	BT_MESH_MODEL_OP_END,
+};
+
+static ssize_t scene_store(struct bt_mesh_model *mod, uint8_t data[])
+{
+	struct bt_mesh_light_xyl_srv *srv = mod->user_data;
+	struct bt_mesh_light_xy_cb_rsp xy_rsp;
+
+	srv->handlers->xy_get(srv, NULL, &xy_rsp);
+
+	if (xy_rsp.remaining_time) {
+		sys_put_le16(xy_rsp.target.x, data);
+		sys_put_le16(xy_rsp.target.y, data);
+	} else {
+		sys_put_le16(xy_rsp.current.x, data);
+		sys_put_le16(xy_rsp.current.y, data);
+	}
```

Comment:
`sys_put_le` does not work the same way as net_buf. You'll have to set the data array index for each

Reply:
`sys_put_le16(xy_rsp.current.x, data);`
`sys_put_le16(xy_rsp.current.y, data + sizeof(uint16_t));`

`.params.x = sys_get_le16(data),`
`.params.y = sys_get_le16(data + sizeof(uint16_t)),`
?